TaskHero/TaskCardCell.swift

```diff
@@ -53,26 +66,95 @@ class TaskCardCell: UITableViewCell {
+        for iconImageView in iconImageViews! {
+            iconImageView.isHidden = true
+            iconImageView.clipsToBounds = true
+            iconImageView.layer.cornerRadius = iconImageView.bounds.width / 2
+        }
+
+        fetchUsers()
     }
     
     @objc fileprivate func onTaskTap(sender: UITapGestureRecognizer) {
-        print("---- on TaskTapped")
         delegate?.taskTapped(self)
     }
     
     @objc fileprivate func onTaskLongPress(sender: UILongPressGestureRecognizer) {
-        print("---- on LongPress")
         delegate?.taskLongPressed(self)
     }
     
     @objc fileprivate func onTaskPan(sender: UIPanGestureRecognizer) {
-        print("---- on Pan")
-        // TODO: add delete task functionality here.
+        let width = bounds.size.width
+        let translation = sender.translation(in: self)
+        let velocity = sender.velocity(in: self)
+        
+        if sender.state == .began {
+            originalCenter = center
+        } else if sender.state == .changed {
+            center.x = originalCenter.x + max(0, translation.x)
+            alpha = 1 - (translation.x / width)
+        } else if sender.state == .ended {
+            if translation.x > (width / 2) && velocity.x > 0 {
+                UIView.animate(withDuration: 0.25, animations: { 
+                    self.center.x = self.center.x + width
+                    self.alpha = 0
+                }, completion: {(complete) in
+                    self.delegate?.taskCellWasRemoved(self)
+                })
+            } else {
+                UIView.animate(withDuration: 0.25, animations: { 
+                    self.center.x = self.originalCenter.x
+                    self.alpha = 1
+                })
+            }
+        }
+    }
+    
+    override func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
+        if let pan = gestureRecognizer as? UIPanGestureRecognizer {
+            let velocity = pan.velocity(in: self)
+            return fabs(velocity.x) > fabs(velocity.y)
+        }
+        
+        return super.gestureRecognizerShouldBegin(gestureRecognizer)
     }
     
     override func setSelected(_ selected: Bool, animated: Bool) {
         super.setSelected(selected, animated: animated)
     }
-
+    
+    func animateBackToOriginalPosition() {
+        UIView.animate(withDuration: 0.25, animations: {
+            self.center.x = self.originalCenter.x
+            self.alpha = 1
+        })
+    }
+    
+    fileprivate func fetchUsers() {
+        ParseClient.sharedInstance.getTeammates(
+            success: { (users) in
+                self.users = users
+                self.setMemberIcons()
+        },
+            failure: { (error) in
+                print("ERROR - failed to fetch users")
+        })
+    }
+    
+    fileprivate func setMemberIcons() {
+        // TODO: load only users who have steps assigned in this Task
```

Comment:
Look at the TaskInstance.getInvolvedUsers() method. You might want to implement the same thing into the Task model too.